File: code/Ch 10/JSTokenScanner/TokenScannerUnitTest.java
/*
 * File: TokenScannerUnitTest.java
 * -------------------------------
 * This program implements a unit test of the simplified version of the
 * TokenScanner class.
 */

package edu.stanford.cs.javacs2.ch7.exercises;

import edu.stanford.cs.javacs2.ch7.TokenScanner;
import static edu.stanford.cs.unittest.UnitTest.assertEquals;

public class TokenScannerUnitTest {

   public void run() {
      TokenScanner scanner = new TokenScanner("This is Pig Latin.");
      assertEquals(scanner.nextToken(), "This");
      assertEquals(scanner.nextToken(), " ");
      assertEquals(scanner.nextToken(), "is");
      assertEquals(scanner.nextToken(), " ");
      assertEquals(scanner.nextToken(), "Pig");
      assertEquals(scanner.nextToken(), " ");
      assertEquals(scanner.nextToken(), "Latin");
      assertEquals(scanner.hasMoreTokens(), true);
      assertEquals(scanner.nextToken(), ".");
      assertEquals(scanner.hasMoreTokens(), false);
      assertEquals(scanner.nextToken(), "");
      scanner.setInput("double area = Math.PI * r * r;");
      scanner.ignoreWhitespace();
      assertEquals(scanner.nextToken(), "double");
      assertEquals(scanner.nextToken(), "area");
      scanner.saveToken("area");
      assertEquals(scanner.nextToken(), "area");
      assertEquals(scanner.nextToken(), "=");
      assertEquals(scanner.nextToken(), "Math");
      assertEquals(scanner.nextToken(), ".");
      assertEquals(scanner.nextToken(), "PI");
      assertEquals(scanner.nextToken(), "*");
      assertEquals(scanner.nextToken(), "r");
      assertEquals(scanner.nextToken(), "*");
      assertEquals(scanner.nextToken(), "r");
      assertEquals(scanner.nextToken(), ";");
      assertEquals(scanner.hasMoreTokens(), false);
   }

/* Main program */

   public static void main(String[] args) {
      new TokenScannerUnitTest().run();
   }

}

File: code/Ch 9/PigLatin/TokenScannerUnitTest.js
/*
 * File: TokenScannerUnitTest.js
 * -----------------------------
 * This file tests the implementation of the simplified TokenScanner class.
 */

"use strict";

function TokenScannerUnitTest() {
   JSUnitTest.resetErrorCount();
   const INPUT = "This is Pig Latin."
   let scanner = TokenScanner(INPUT);
   JSUnitTest.assertTrue("scanner.hasMoreTokens() -> " +
                         scanner.hasMoreTokens(),
                         scanner.hasMoreTokens());
   let token = scanner.nextToken();
   JSUnitTest.assertEquals("scanner.nextToken() -> \"" + token + "\"",
                           token, "This");
   token = scanner.nextToken()
   JSUnitTest.assertEquals("scanner.nextToken() -> \"" + token + "\"",
                           token, " ");
   token = scanner.nextToken()
   JSUnitTest.assertEquals("scanner.nextToken() -> \"" + token + "\"",
                           token, "is");
   token = scanner.nextToken()
   JSUnitTest.assertEquals("scanner.nextToken() -> \"" + token + "\"",
                           token, " ");
   token = scanner.nextToken()
   JSUnitTest.assertEquals("scanner.nextToken() -> \"" + token + "\"",
                           token, "Pig");
   token = scanner.nextToken()
   JSUnitTest.assertEquals("scanner.nextToken() -> \"" + token + "\"",
                           token, " ");
   token = scanner.nextToken()
   JSUnitTest.assertEquals("scanner.nextToken() -> \"" + token + "\"",
                           token, "Latin");
   token = scanner.nextToken()
   JSUnitTest.assertEquals("scanner.nextToken() -> \"" + token + "\"",
                           token, ".");
   JSUnitTest.assertFalse("scanner.hasMoreTokens() -> " +
                          scanner.hasMoreTokens(),
                          scanner.hasMoreTokens());
   token = scanner.nextToken()
   JSUnitTest.assertEquals("scanner.nextToken() -> \"" + token + "\"",
                           token, "");
   if (JSUnitTest.getErrorCount() === 0) {
      console.println("TokenScannerUnitTest succeeded");
   } else {
      console.println("TokenScannerUnitTest failed");
   }
}
